### PR TITLE
Add docs / table for Prometheus endpoint.

### PR DESCRIPTION
**Pull request by benarent**

### CODE REVIEW

File: docs/4.3/metrics_logs_reference.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Metrics
+
+## Teleport Prometheus Endpoint
+
+Teleport provides HTTP endpoints for monitoring purposes. They are disabled
+by default, but you can enable them:
```
Comment (webvictim) [inline docs/4.3/metrics_logs_reference.md:6]: "enable them:" -> "enable them using the `--diag-addr` flag to `teleport start`:"

File: docs/4.3/metrics_logs_reference.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+$ teleport start --diag-addr=127.0.0.1:3000
+```
+
+Now you can see the monitoring information by visiting several endpoints:
+
+* `http://127.0.0.1:3000/metrics` is the list of internal metrics Teleport is
+   tracking. It is compatible with [Prometheus](https://prometheus.io/)
+   collectors.
+
+<!-- Help: Edit this table https://docs.google.com/spreadsheets/d/1lYEGjENdsEoMXGg6LO71-U5Y6PDzPbplut255aacGWo/edit#gid=0 and put the table into https://www.tablesgenerator.com/markdown_tables  -->
```
Comment (webvictim) [inline docs/4.3/metrics_logs_reference.md:18]: What's this comment about?

**Review by webvictim — COMMENTED** [review]
Looks OK

File: assets/monitoring/README.md [hunk 3]
```diff
@@ -29,30 +30,41 @@ python convert.py health-raw.json health-dashboard.json
 
 ## Low level monitoring
 
-Teleport adds `gops` as a low level debugging solution:
+Teleport can be started with Golang's standard profiler.
```
Comment (awly) [inline assets/monitoring/README.md:33]: nit: `s/Golang/Go/g`

Comment (awly) [inline assets/monitoring/README.md:33]: how about a link to https://golang.org/pkg/net/http/pprof/ here?

File: assets/monitoring/README.md [hunk 4]
```diff
@@ -29,30 +30,41 @@ python convert.py health-raw.json health-dashboard.json
...
+Goroutine profile shows how many concurrent Golang “lightweight threads” are used
+in the system:
+
+`curl -o goroutine.profile http://127.0.0.1:3000/debug/pprof/goroutine`
+
+Heap profile shows allocated objects in the system:
+
+`curl -o heap.profile http://127.0.0.1:3000/debug/pprof/heap`
+
+The resulting profiles go tool pprof tool:
```
Comment (awly) [inline assets/monitoring/README.md:62]: "To view the resulting profiles, use go tool pprof:"

File: assets/monitoring/teleport.service [hunk 5]
```diff
@@ -7,7 +7,7 @@ Type=simple
 Restart=always
 User=ubuntu
 LimitNOFILE=32000
-ExecStart=/home/ubuntu/go/src/github.com/gravitational/teleport/build/teleport start --config=/etc/teleport/teleport.yaml  --gops --gops-addr=127.0.0.1:4321 --diag-addr=127.0.0.1:3434
+ExecStart=/home/ubuntu/go/src/github.com/gravitational/teleport/build/teleport start --config=/etc/teleport/teleport.yaml  -d --diag-addr=127.0.0.1:3434
```
Comment (awly) [inline assets/monitoring/teleport.service:10]: change port number here too

File: docs/4.3/metrics_logs_reference.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+## Teleport Prometheus Endpoint
+
+Teleport provides HTTP endpoints for monitoring purposes. They are disabled
+by default, but you can enable them:
+
+```bash
+$ teleport start --diag-addr=127.0.0.1:3000
+```
+
+Now you can see the monitoring information by visiting several endpoints:
```
Comment (awly) [inline docs/4.3/metrics_logs_reference.md:12]: This says "several endpoints" but only mentions one

File: docs/4.3.yaml [hunk 7]
```diff
@@ -86,4 +86,5 @@ nav:
     - Reference:
         - YAML: config_reference.md
         - CLI: cli-docs.md
-        - API: api_reference.md
\ No newline at end of file
+        - API: api_reference.md
+        - Metrics & Logs: metrics_logs_reference.md
```
Comment (awly) [inline docs/4.3.yaml:90]: nit: we don't have info about logs on that page yet

File: assets/monitoring/teleport.service (hunk 5 above)
Comment (benarent) [reply assets/monitoring/teleport.service:10]: I was going to add a note, apart from legacy reasons.. any reason why we still running the marketplace on 3434? https://github.com/gravitational/teleport/search?q=3434&unscoped_q=3434 @webvictim

**Comment by benarent** [comment]
@fspmarshall PTAL ~ RJ said you've been working on performance related things.

File: docs/4.3/metrics_logs_reference.md (hunk 2 above)
Comment (benarent) [reply docs/4.3/metrics_logs_reference.md:18]: Creating these tables is a bit of a pain, I think long term we should automate it but didn't want to put the old optimisation horse before the cart.  This makes bulk changes much easier.

File: assets/monitoring/teleport.service (hunk 5 above)
Comment (benarent) [reply assets/monitoring/teleport.service]: I won't change the marketplace image in this PR but we should come back and review it. Maybe look at shipping these to CloudWatch.

File: docs/4.3.yaml (hunk 7 above)
Comment (benarent) [reply docs/4.3.yaml]: Removed from sidebar for now.

**Review by fspmarshall — APPROVED** [review]
LGTM.  One potential nicety would be to mention the `--web` flag on `pprof`, which is awesome.  Ex:

```
go tool pprof --web heap.profile
```

^^^ Opens a nice SVG diagram in the browser which makes it really easy to get a high-level view of whats going on.

**Comment by benarent** [comment]
PTAL @webvictim

File: assets/monitoring/README.md [hunk 8]
```diff
@@ -6,13 +6,14 @@ Teleport exports Prometheus Metrics endpoints.
...
-teleport start --diag-addr=127.0.0.1:3434
+```bash
+teleport start --diag-addr=127.0.0.1:3000
+# http://127.0.0.1:3000/metrics
 ```
 
 To start monitoring stack, simply `docker-compose up`
 
-Then go to `http://localhost:3000`
+Then go to `http://localhost:3001`
```
Comment (webvictim) [inline assets/monitoring/README.md:16]: Why a different port in `docker-compose` versus running directly on the machine?

**Review by webvictim — APPROVED** [review]
LGTM, one nit

File: assets/monitoring/README.md (hunk 8 above)
Comment (benarent) [reply assets/monitoring/README.md:16]: I did change it, https://github.com/gravitational/teleport/pull/4069/files#diff-f15524a82953c688d5d2f7086398011eR24 I did this to run Teleport Diagnostics on it's default port; when running locally, but let me know if there is a better way to do this.

**Comment by benarent** [comment]
Sorry @webvictim I thought I needed another review but I had all the reviews needed.

**Pull request merged**